### PR TITLE
Create a helper template for stable random values

### PR DESCRIPTION
**Pull request by acohen4**
In this solution, the current timestamp is used to compute a hash. 

There's an obvious, though probabilistically unlikely problem of having the time change between invocations of this template within the script, where the resulting hash wouldn't be stable.

### CODE REVIEW

**Comment by acohen4** [comment]
closes https://github.com/pachyderm/pachyderm/issues/6660

File: etc/helm/pachyderm/templates/_helpers.tpl [hunk 1]
```diff
@@ -24,13 +24,28 @@ LOCAL
...
+## truncate the date to the minute and use 'genPrefix' to produce a different hash per useage
+## expects a context containing 'defaultValue' and 'hashSalt' keys
+{{- define "defaultOrStableHash" -}}
+{{- if .defaultValue }}
+    {{- .defaultValue }}
+{{- else }}
+    {{- if .OnlyDefault }}
+        {{- fail "must provide a default value during release upgrade." }}
+    {{- else }}
+        {{- derivePassword 1 "long" (now | toString | trunc 16) .hashSalt "pachyderm" -}}
```
Comment (chainlink) [inline etc/helm/pachyderm/templates/_helpers.tpl:48]: Can you add some docs here? (ie, talking about the date truncation, perhaps referencing the article we found this in.

File: etc/helm/pachyderm/templates/_helpers.tpl [hunk 2]
```diff
@@ -24,13 +24,44 @@ LOCAL
...
+## is because we want to this template to produce the same hash each 
+## time it's executed in a release. In this case, the current timestamp 
+## gives us a unique enough value to use as a hash seed. We truncate it 
+## to the minute to reduce the odds that it will produce different timestamp 
+## values between renderings of this template.
+## Reference https://github.com/helm/helm/issues/6456
+##
+## 2.) 'onlyDefault' is used to force this template to fail if a default value is not provided
+{{- define "defaultOrStableHash" -}}
+{{- if .defaultValue }}
```
Comment (chainlink) [inline etc/helm/pachyderm/templates/_helpers.tpl:58]: This method might be a little "too smart" for helm (specifically passing in a hash as an arg) Generally with helm it's ok to break DRY to make things more clear / use values directly.

**Review by chainlink — CHANGES_REQUESTED** [review]
This looks good, but we need a couple of changes on the testing side.

I think we should ship this PR, but the more I think about it, I think we should fix the root issue in the config pod before RC (being able to reference the secret values from Env vars/ secrets instead of embedding them in the config)

**Pull request closed without merging**